### PR TITLE
support data-loading="lazy|eager" for lazy loading

### PR DESCRIPTION
**Pull request by dmanek**
Add support for lazy loading using `data-loading` attribute. The following components are addressed in this PR:
1. amp-twitter
2. amp-facebook
4. amp-video-iframe
5. amp-youtube

### CODE REVIEW

File: extensions/amp-video-iframe/1.0/base-element.js [hunk 1]
```diff
@@ -15,6 +15,7 @@ BaseElement['props'] = {
   'poster': {attr: 'poster'},
   'src': {attr: 'src'},
   'controls': {attr: 'controls', type: 'boolean'},
+  'loading': {attr: 'data-loading'},
```
Comment (alanorozco) [inline extensions/amp-video-iframe/1.0/base-element.js:18]:
```suggestion
  'loading': {attr: 'data-loading', default: 'lazy'},
```

File: extensions/amp-twitter/1.0/base-element.js [hunk 2]
```diff
@@ -9,7 +9,7 @@ BaseElement['Component'] = BentoTwitter;
 /** @override */
 BaseElement['props'] = {
   'title': {attr: 'title'}, // Needed for Preact component
-  'options': {attrPrefix: 'data-'}, // Needed to render componoent upon mutation
+  'options': {attrPrefix: 'data-'}, // Needed to render component upon mutation
```
Comment (alanorozco) [inline extensions/amp-twitter/1.0/base-element.js:12]: Kind of odd to make this part of `options` in `<ProxyIframeEmbed />`. Would be simpler if it's part of `{...rest}` to propagate `loading` to the leaf iframe.

Does this work? It's okay if we pass an extraneous `options.loading` for now.

```suggestion
  'loading': {attr: 'data-loading'},
  'options': {attrPrefix: 'data-'}, // Needed to render component upon mutation
```

File: extensions/amp-video/1.0/storybook/VideoIframe.js [hunk 3]
```diff
@@ -62,6 +62,7 @@ const AmpVideoIframeLike = ({unloadOnPause, ...rest}) => {
       allow="autoplay" // this is not safe for a generic frame
       onMessage={onMessage}
       makeMethodMessage={makeMethodMessage}
+      data-loading="lazy"
```
Comment (alanorozco) [inline extensions/amp-video/1.0/storybook/VideoIframe.js:65]: Is this necessary? Should we set the default at the `<VideoIframe />` level instead?

File: src/preact/component/3p-frame.js [hunk 4]
```diff
@@ -160,6 +160,7 @@ function ProxyIframeEmbedWithRef(
       src={src}
       title={title}
       {...rest}
+      loading={options.loading ?? 'lazy'}
```
Comment (alanorozco) [inline src/preact/component/3p-frame.js:163]: Ditto propagating by `{...rest}`

**Review by alanorozco — CHANGES_REQUESTED** [review]

File: extensions/amp-twitter/1.0/base-element.js (hunk 2 above)
Comment (dmanek) [reply extensions/amp-twitter/1.0/base-element.js:13]: Yes, including `'loading': {attr: 'data-loading'}` works. Updated.

File: extensions/amp-video/1.0/storybook/VideoIframe.js (hunk 3 above)
Comment (dmanek) [reply extensions/amp-video/1.0/storybook/VideoIframe.js]: Removed. Remnant of testing.

File: src/preact/component/3p-frame.js (hunk 4 above)
Comment (dmanek) [reply src/preact/component/3p-frame.js]: Done.

**Review by alanorozco — APPROVED** [review]
Let's document the attribute on each component pls

**Comment by dmanek** [comment]
Per offline conversation, will update documentation once all iframe based components are completed.

**Pull request merged**